File: Admin/Admin.Server/RQ/Query/AllOrgRQ.cs
using com.etsoo.CoreFramework.Application;
using com.etsoo.Utils.Actions;

namespace Admin.Server.RQ.Query
{
    /// <summary>
    /// Query all organizations request data
    /// 查询所有机构请求数据
    /// </summary>
    public record AllOrgRQ : QueryIntRQ
    {
        /// <summary>
        /// Parent org. ID
        /// 父机构编号
        /// </summary>
        public int? ParentId { get; init; }

        /// <summary>
        /// Owner id
        /// 所有人编号
        /// </summary>
        public int? OwnerId { get; init; }

        /// <summary>
        /// PIN
        /// 唯一编号
        /// </summary>
        public string? Pin { get; init; }

        /// <summary>
        /// Creation start
        /// 登记开始时间
        /// </summary>
        public DateTime? CreationStart { get; init; }

        /// <summary>
        /// Creation end
        /// 登记结束时间
        /// </summary>
        public DateTime? CreationEnd { get; init; }

        public override IActionResult? Validate()
        {
            var result = base.Validate();
            if (result != null)
            {
                return result;
            }

            if (Pin != null && Pin.Length > 20)
            {
                return ApplicationErrors.NoValidData.AsResult(nameof(Pin));
            }

            return null;
        }
    }
}

File: Admin/admin.client/src/api/rq/query/AllOrgRQ.ts
import { QueryRQ } from "@etsoo/appscript";

/**
 * Query all organizations request data
 * 查询所有机构请求数据
 */
export type AllOrgRQ = QueryRQ & {
  /**
   * Parent org. ID
   * 父机构编号
   */
  parentId?: number;

  /**
   * Owner id
   * 所有人编号
   */
  ownerId?: number;

  /**
   * PIN
   * 唯一编号
   */
  pin?: string;

  /**
   * Creation start
   * 登记开始时间
   */
  creationStart?: Date;

  /**
   * Creation end
   * 登记结束时间
   */
  creationEnd?: Date;
};
